crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2346,84 +2428,238 @@ impl<'src> Parser<'src> {
         }
     }
 
-    /// Parses a parameter list.
+    /// Parses a parameter list for the given function kind.
     ///
     /// See: <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-python-grammar-parameter_list>
     pub(super) fn parse_parameters(&mut self, function_kind: FunctionKind) -> ast::Parameters {
-        let mut args = vec![];
-        let mut posonlyargs = vec![];
-        let mut kwonlyargs = vec![];
-        let mut kwarg = None;
-        let mut vararg = None;
+        let start = self.node_start();
 
-        let mut has_seen_asterisk = false;
-        let mut has_seen_vararg = false;
-        let mut has_seen_default_param = false;
+        // TODO(dhruvmanila): This has the same problem where if there are multiple
+        // kwarg or vararg, the last one will win and the parser will drop the
```

Comment:
It would help if you can rephare the "same problem". It's unclear what it refers to. Is it arguments parsing?